src/tools/server-autosave.js

```diff
@@ -0,0 +1,18 @@
+import {
+  setPreviouslySavedJsmdToCurrentJsmd,
+  saveNotebookToServer
+} from "../actions/actions";
+
+export function changeMadeToJsmd(state) {
+  return state.previouslySavedJsmd !== state.jsmd;
+}
+export function checkForServerAutosave(store) {
+  store.dispatch(setPreviouslySavedJsmdToCurrentJsmd());
+  setInterval(() => {
+    const state = store.getState();
+    if (changeMadeToJsmd(state)) {
+      store.dispatch(saveNotebookToServer(false));
+      store.dispatch(setPreviouslySavedJsmdToCurrentJsmd());
```

Comment:
edge case, but i wonder if perhaps `setPreviouslySavedJsmdToCurrentJsmd()` should be called within the `then` of the successful resolution of the `saveNotebookRequest` call _inside of_ `saveNotebookToServer(false)`.

this really is a small detail, but i think that maybe as written, the `store.dispatch(saveNotebookToServer(false));` could kick of it's server request, and we'd still dispatch the `store.dispatch(setPreviouslySavedJsmdToCurrentJsmd());` even if the request failed for some reason. (also possible very minor race condition if there are changes made while the server request is in flight)

not sure it matters, but it just seems more correct somehow to set the previously saved contents within the confirmation from the server that the save request was successful

Reply:
I think this is the right idea. We need a Promise to be returned here so we can catch errors of various kinds.